tests/testthat/test-prep_data.R

```diff
@@ -410,6 +410,12 @@ test_that("add_levels doesn't add levels to outcome", {
   expect_false(any(c("other", "hcai_missing") %in% levels(pd$x)))
 })
 
+test_that("prep_data respects add_levels = FALSE", {
+  d <- data.frame(x = c("A", "B"), y = 1:2)
+  pd <- prep_data(d, add_levels = FALSE, make_dummies = FALSE, impute = FALSE)
+  expect_false(any(c("other", "hcai_missing") %in% levels(pd$x)))
```

Comment:
I may be doing something wrong, but I can't get this test to fail, and I suspect it never could.

The following snippet returns false, when I believe the intention for the test is that it should return true.

```
d <- data.frame(x = c("A", "B"), y = 1:2)
any(c("A")) %in% levels(d$x)
```

The fix should be a simple as moving a parenthesis around the `any` to the end of the line.

If this test were written first before writing the fix (TDD), it would have been immediately apparent that it was passing, and therefore a bad test.

Reply:
This test works, you're just testing the input data.frame (which does always have level "A" in "x") instead of the output data.frame. `any(c("A", "other") %in% levels(pd$x))` is TRUE.

Reply:
You also have the closing paren too early in your code snippet. In the test, `any` ends after `levels`.